salt/modules/kubernetes.py

```diff
@@ -0,0 +1,500 @@
+'''
+Module for handling kubernetes calls.
+
+:optdepends:    - kubernetes Python client
+:configuration: This module is not usable until the following are specified
+    either in a pillar or in the minion's config file::
+
+        kubernetes.user: admin
+        kubernetes.password: verybadpass
+        kubernetes.api_url: 'http://127.0.0.1:8080'
+'''
+# Import Python Futures
+from __future__ import absolute_import
+from six import iteritems
+from salt.exceptions import CommandExecutionError
+import salt.utils
+import salt.utils.templates
+import yaml
+
+try:
+    import kubernetes
+    import kubernetes.client
+    from kubernetes.client.rest import ApiException
+
+    HAS_LIBS = True
+except ImportError:
+    HAS_LIBS = False
+
+
+import logging
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'kubernetes'
+
+
+def __virtual__():
+    '''
+    Check dependencies
+    '''
+    if HAS_LIBS:
+        return __virtualname__
+
+    return False
+
+
+def ping():
+    '''
+    Checks connections with the kubernetes API server.
+    Returns True if the connection can be established, False otherwise.
+
+    CLI Example:
+        salt '*' kubernetes.ping
+    '''
+    status = True
+    try:
+        nodes()
+    except:
```

Comment:
We prefer not to have bare exceptions. We prefer `except Exception` here, which should catch all cases.